frontend/src/stimulus/controllers/poll-for-changes.controller.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * -- copyright
+ * OpenProject is an open source project management software.
+ * Copyright (C) 2023 the OpenProject GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 3.
+ *
+ * OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+ * Copyright (C) 2006-2013 Jean-Philippe Lang
+ * Copyright (C) 2010-2013 the ChiliProject Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * See COPYRIGHT and LICENSE files for more details.
+ * ++
+ */
+
+import { ApplicationController } from 'stimulus-use';
+import { renderStreamMessage } from '@hotwired/turbo';
+
+export default class PollForChangesController extends ApplicationController {
+  static values = {
+    url: String,
+    interval: Number,
+    reference: String,
+  };
+
+  declare referenceValue:string;
+  declare urlValue:string;
+  declare intervalValue:number;
+
+  private interval:number;
+
+  private pollListener:() => unknown = () => this.triggerTurboStream();
```

Comment:
Why not use `triggerTurboStream` directly?

Reply:
You need to have the function bound as a variable so that it can be correctly remove it. Just calling `removeEventListener('..', this.myFunction.bind(this)` will not work, as `.bind` returns a new function instance.

Reply:
I only understood now that `pollListener` captures `this` because it is an arrow function